src/app/auth/login.ts

```diff
@@ -54,23 +54,65 @@ export const loginWithPhone = async ({
 
   const authService = AuthWithPhonePasswordlessService()
 
-  let kratosResult = await authService.login(phone)
-
+  let kratosResult = await authService.loginToken(phone)
   // FIXME: this is a fuzzy error.
   // it exists because we currently make no difference between a registration and login
   if (kratosResult instanceof LikelyNoUserWithThisPhoneExistError) {
     // user is a new user
-
     kratosResult = await authService.createIdentityWithSession(phone)
     if (kratosResult instanceof Error) return kratosResult
     addAttributesToCurrentSpan({ "login.newAccount": true })
   } else if (kratosResult instanceof Error) {
     return kratosResult
   }
-
   return kratosResult.sessionToken
 }
 
+export const loginWithPhoneCookie = async ({
+  phone,
+  code,
+  ip,
+}: {
+  phone: PhoneNumber
+  code: PhoneCode
+  ip: IpAddress
+}): Promise<WithCookieResponse | ApplicationError> => {
+  {
+    const limitOk = await checkFailedLoginAttemptPerIpLimits(ip)
+    if (limitOk instanceof Error) return limitOk
+  }
+
+  {
+    const limitOk = await checkFailedLoginAttemptPerPhoneLimits(phone)
+    if (limitOk instanceof Error) return limitOk
+  }
+
+  // TODO:
```

Comment:
Is this the right place for this?

Reply:
i copied that code from @nicolasburtey loginToken. will defer to him